teps/0070-tekton-catalog-task-platform-support.md

```diff
@@ -0,0 +1,129 @@
+---
+status: proposed
+title: Platform support in Tekton catalog
+creation-date: '2021-06-02'
+last-updated: '2021-06-02'
+authors:
+- '@barthy1'
+---
+
+# TEP-0070: Platform support in Tekton catalog
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Catalog side](#catalog-side)
+  - [Hub side](#hub-side)
+  - [Tekton CLI side](#tekton-cli-side)
+- [References (optional)](#references-optional)
+
+## Summary
+
+This TEP proposes to add information about supported operating systems/hardware architectures (platforms) to the Tekton catalog resources and provide ability to the user to filter/search resources from the catalog for specific platform.
+
+## Motivation
+
+Tekton at this moment has releases for 4 hardware architectures(platforms) - `linux/amd64`, `linux/s390x`, `linux/ppc64le` and `linux/arm64`. Tekton catalog contains already great list of resources to run. However there is no information about which resource can be executed on which platform. As a result user can fail running a task on top of not supported hardware architecture.
+
+### Goals
+
+- Define the way to add supported platforms into resource definition in the Tekton catalog.
+- Define the ways to show information about platforms to the user.
+
+### Non-Goals
+
+- Define modifications in Tekton to better support running tasks on different platforms ([TEP-0041](https://github.com/tektoncd/community/pull/310)).
+- Limit/hide ("no-show" or "no-access") catalog resources based on host platform.
+
+### Use Cases (optional)
+
+- Filter the resources in the Hub UI using platforms.
+- Search the resources with `tkn hub` for specific platform(s).
+- Get information about supported platforms for specific resource in the Hub UI/`tkn hub`.
+
+## Requirements
+
+- Platform information in the catalog resource should be optional.
+- Information how to identify/prove supported platform for the resource should be documented in the Tekton catalog.
+- Tekton Hub should automatically get/update information about platforms of the resources.
+- Hub UI should show all platforms available in database to filter.
+
+## Proposal
+
+### Catalog side
+
+Add new `tekton.dev/platforms` annotation to the resource yaml file with list of platforms separated by comma, for example
+
+```
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: image-build
+    tekton.dev/platforms: linux/amd64,linux/s390x
+```
+
+The `tekton.dev/platforms` annotation should be optional. 
```

Comment:
If this is not specified, what it will mean like

1. If annotation is not there then it means all platforms
2. If annotation is not there then it means any specific platforms like linux/amd64

Reply:
If annotation is not specified, it means `linux/amd64` only.

Reply:
So in that case I think `tekton.dev/annotation` should be made mandatory

Reply:
From my point of view - it is not required, if 
- there is an agreement, that lack of annotation means `linux/amd64`. Which is actually true in reality, because all currently available tasks are executable on`linux/amd64` and people who submitted task's information tested the task most probably only for `linux/amd64`.
- Hub parser will transform no `tekton.dev/platforms` annotation to `tekton.dev/platforms: linux/amd64` annotation on its side

This approach will provide backward compatibility + represent current state of the tasks.

Reply:
@barthy1 does that mean all new task can be assummed to executable on `linux/amd64` by default?
for eg. if author has added `tekton.dev/platforms: linux/s390x` so in this it should be assumed for `linux/amd64` also? or author has to explicitly add it?

Reply:
That's my point. If author adds information about platform (for instance `tekton.dev/platforms: linux/s390x`), no extra platforms should be added. The list of supported platforms will be exactly what is mentioned in `tekton.dev/platforms`.
But in case if author doesn't provide any information about platforms, the default value is `linux/amd64`, as this platform is actually default one in cloud/container world

Reply:
> This approach will provide backward compatibility + represent current state of the tasks.

I agree with this from a developer point of view.

From an enduser point of view, I would prefer to see the annotation on all tasks even when only `linux/amd64` is supported. The is reason is that the lack of annotation can be intrepreted as
- multi-arch support - unknown
- multi-arch suppot - none

Reply:
@nikhil-thomas agree with what you said. And it also makes sense too me to add `linux/amd64` even if it's only one platform. I do not propose skip the annotation in case if only `linux/amd64` is supported. 
But I still don't think that `tekton.dev/platforms` should be mandatory, as it's very strict requirement, which   
does not allow to use current tasks(and their not latest versions) at all.
Can we say that providing `tekton.dev/platforms` information should be best practice and recommended?